src/helpers/buildPaperslistByContact.js

```diff
@@ -3,23 +3,43 @@ import { isReferencedById } from 'cozy-client/dist/associations'
 import { CONTACTS_DOCTYPE } from 'src/doctypes'
 const { getFullname } = models.contact
 
+const DEFAULT_MAX_DISPLAY = 3
+
+/**
+ * @typedef {object} BuildPaperslistByContactParam
+ * @property {object[]} papersList - Array of IOCozyFile
+ * @property {object[]} contactsList - Array of cozy contact object
+ * @property {Paper[]} [papersDefinitions=[]] Array of Papers
+ * @property {string} [currentFileCategory=''] - Category of qualification
+ */
+
 /**
- * @param {IOCozyFile[]} papersList - An array of IOCozyFile
- * @param {object[]} contactsList - An array of cozy contact object
- * @returns {{ contact: string, papers: IOCozyFile[] }[]}
+ * @param {BuildPaperslistByContactParam} param
+ * @returns {{ contact: string, papers: {maxDisplay: number, list: IOCozyFile[]} }[]}
  */
-export const buildPaperslistByContact = (papersList, contactsList) => {
-  const result =
-    Array.isArray(contactsList) && Array.isArray(papersList)
-      ? contactsList.flatMap(contact => [
-          {
-            contact: getFullname(contact),
-            papers: papersList.filter(paper =>
-              isReferencedById(paper, CONTACTS_DOCTYPE, contact._id)
-            )
-          }
-        ])
-      : []
+export const buildPaperslistByContact = ({
+  papersList,
+  contactsList,
+  papersDefinitions = [],
+  currentFileCategory = ''
+}) => {
```

Comment:
Do we really have all the papers already loaded? Shouldn't we make request to fetch them by contacts with a limit? And then call the fetchMore method? 

We've a loadMore component that handle this kind of thing https://docs.cozy.io/cozy-ui/react/#!/LoadMore can we use it?

Reply:
Approche intéressante, si je comprends bien :
1. Récupérer les contacts (avec une limite)
2. Faire une requête qui récupère tous les fichiers faisant référence à au moins un des contacts

Le problème (tout comme actuellement d'ailleurs), est que certains fichiers (venant d'un Connector ou Drive), n'ont pas de contact

Il ya une carte Trello remontant ce bug, je suis en train de m'en charger en parallèle mais ça sera sur une autre PR

Reply:
Non c'est pas ça, on en parlera à l'occasion :)

Reply:
Ok 👍